src/middlewared/middlewared/apidocs/templates/websocket/query.md

```diff
@@ -0,0 +1,131 @@
+## Query Methods
+
+TrueNAS API has multiple query methods including `pool.query`, `disk.query`, `vm.query`, and many more.
+
+The arguments for these methods support multiple options and filters that are similar to SQL queries.
+
+### Query Filters
+
+#### Basic Usage
+
+Query Filters are primarily an array of conditions, with each condition also represented as an array.
+
+Each condition in the filter list should compare a field with a value.
+
+eg. Filter Syntax: `["field", "operator", value]` 
+
+For example, to filter the data returned by `disk.query`, we provide a list of conditions:
+
+Javascript:
+    :::javascript
+    [
+      ["name","=","ada1"] 
+    ]
+
+Command Line: `# midclt call disk.query '[["name", "=", "ada1"]]'`
+
+NOTE: Supported Operators include `'=' | '!=' | '>' | '>=' | '<' | '<=' `
+
+#### Multiple Filters
+
+We can use `disk.query` with the "type" and "rotationrate" filters to find hard drives with a rotation rate higher than 5400 RPM:
+
+Javascript:
+    :::javascript
+    [
+      ["type","=","HDD"],
+      ["rotationrate",">",5400] // Note that the value should be the correct type
+    ]
+
+Command Line: `# midclt call disk.query '[["type", "=", "HDD"],["rotationrate",">",5400]]'`
```

Comment:
Please remove all command line mentions. We don't intend to teach developers to use midclt